scipy/optimize/_differentialevolution.py

```diff
@@ -1298,7 +1299,10 @@ def fun(x):
                 return np.atleast_1d(constraint.fun(x))
         elif isinstance(constraint, LinearConstraint):
             def fun(x):
-                A = np.atleast_2d(constraint.A)
+                if not issparse(constraint.A):
```

Comment:
Could we do this instead, because it makes it clearer what to do if it's sparse:

```
if issparse(constraint.A):
    A = constraint.A
else:
    A = np.atleast_2d(constraint.A)
```

Reply:
Do these changes work with all kinds of sparse matrices? Do all sparse matrices have an `A` attribute?
e.g. dok, lil, bsr, etc

Reply:
The changes work with all kinds of sparse matrices, I did not add all to the test to avoid cluttering them, but I can do so and it's easy if you want.
All `LinearConstraints` have an `A` attribute.

Reply:
All scipy.sparse matrices have an `.A` attribute, which is just a shorter way of writing the `.toarray()` method. It's definitely confusing that the `LinearOperator` A attribute is distinct from the sparse matrix A attribute, though.